aya/src/maps/mod.rs

```diff
@@ -243,19 +241,226 @@ fn maybe_warn_rlimit() {
     }
 }
 
+/// eBPF map types.
+#[derive(Debug)]
+pub enum Map {
+    /// A ['Array`] map
+    Array(MapData),
+    /// A [`PerCpuArray`] map
+    PerCpuArray(MapData),
+    /// A [`ProgramArray`] map
+    ProgramArray(MapData),
+    /// A [`HashMap`] map
+    HashMap(MapData),
+    /// A ['PerCpuHashMap'] map
+    PerCpuHashMap(MapData),
+    /// A [`PerfEventArray`] map
+    PerfEventArray(MapData),
+    #[cfg(feature = "async")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "async")))]
+    /// A [`AsyncPerfEventArray`] map
+    AsyncPerfEventArray(MapData),
```

Comment:
I don't think this should be part of the `Map` enum.
Referring to the `parse_map` code:
```rust
        BPF_MAP_TYPE_PERF_EVENT_ARRAY | BPF_MAP_TYPE_LRU_PERCPU_HASH => {
            Ok(Map::PerfEventArray(map))
        }
 ```
 
 You can't tell from a BTF map, `bpf_map_def` or `bpf_map_info` whether a map is supposed to be Async or not since it's a user preference for how they wish to interact with a map from `aya`. You should only have `Map::PerfEventArray` variant and allow that to be converted into either a `map::AsyncPerfEventArray` and `map::PerfEventArray`

Reply:
Agreed I think this makes more sense... I'll just have to implement a try-from function for that conversion, I also wonder if a "chained try-from" will work? 

i.e still 

`mut logs: AsyncPerfEventArray<_> = bpf.take_map("AYA_LOGS")?.try_into()?;` 

Or if it will need to be 

`mut logs: AsyncPerfEventArray<_> = bpf.take_map("AYA_LOGS")?.try_into().try_into?;`

Reply:
Ah Fixed rather seamlessly with some fun macros 

```
impl_try_from_map!(
    ProgramArray from Map::ProgramArray,
    SockMap from Map::SockMap,
    PerfEventArray from Map::PerfEventArray,
    StackTraceMap from Map::StackTraceMap,
);

#[cfg(feature = "async")]
#[cfg_attr(docsrs, doc(cfg(feature = "async")))]
impl_try_from_map!(
    AsyncPerfEventArray from Map::PerfEventArray,
);
```